mmpy_bot/event_handler.py

```diff
@@ -20,27 +19,32 @@ def __init__(
         self,
         driver: Driver,
         settings: Settings,
-        plugins: Sequence[Plugin],
+        plugins: PluginManager,
         ignore_own_messages=True,
     ):
         """The EventHandler class takes care of the connection to mattermost and calling
         the appropriate response function to each event."""
         self.driver = driver
         self.settings = settings
         self.ignore_own_messages = ignore_own_messages
-        self.plugins = plugins
+        self.plug_manager = plugins
 
         self._name_matcher = re.compile(rf"^@?{self.driver.username}\:?\s?")
 
         # Collect the listeners from all plugins
         self.message_listeners = defaultdict(list)
         self.webhook_listeners = defaultdict(list)
-        for plugin in self.plugins:
+
+        for plugin in self.plug_manager:
             for matcher, functions in plugin.message_listeners.items():
                 self.message_listeners[matcher].extend(functions)
             for matcher, functions in plugin.webhook_listeners.items():
                 self.webhook_listeners[matcher].extend(functions)
 
+        # PluginManager also has listeners for "help"
+        for matcher, functions in self.plug_manager.message_listeners.items():
+            self.message_listeners[matcher].extend(functions)
```

Comment:
I know this is how I already did it, but since any message listener needs to be defined on a Plugin and you've now created a nice place to handle all Plugin logic, maybe it makes more sense not to have a `self.message_listeners` on the `EventHandler` at all? We can simply forward any message to the PluginManager and it can then take the appropriate action itself.

Reply:
I tried implementing this on my first try but there was some code that needed to access `message_listeners` when only `EventHandler` was available. Need to give it another try now that I understand the code better.